run-build

```diff
@@ -101,49 +94,53 @@ class ImageBuildRoot(object):
         #         shutil.rmtree(aptcache_dir)
         # os.makedirs(os.path.join(aptcache_dir), exist_ok=True)
 
-        # Generate a sources list
-        sources_args = {
-            'url': repo,
-            'codename': codename,
-            'components': ' '.join(components),
-            'base': base,
-        }
-        sources = dedent('''\
-        deb {url} {codename} {components}
-        deb http://deb.debian.org/debian {base} main contrib non-free
-        deb http://deb.debian.org/debian {base}-updates main contrib non-free
-        deb http://security.debian.org {base}-security main contrib non-free
-        '''.format(**sources_args))
-
-        sources_path = os.path.join(config['build']['tmpdir'],
-                                    'buildroot-sources.list')
-        with open(sources_path, 'w') as f:
-            f.write(sources)
-
-        log.info('Creating buildroot in %s', self.builddir)
-        setup_script = os.path.join(config['build']['helpersdir'],
-                                    'mmdebstrap-setup')
-        cleanup_script = os.path.join(config['build']['helpersdir'],
-                                      'mmdebstrap-cleanup')
-        buildroot_env = os.environ.copy()
-        buildroot_env.update({
-            'BUILDROOT_KEYRING': keyring,
-        })
-        buildroot_cmd = (
-            'mmdebstrap',
-            '--verbose',
-            '--architectures=' + arch,
-            '--variant=minbase',
-            '--include=' + ','.join(packages),
-            '--setup-hook=' + setup_script,
-            '--customize-hook=' + cleanup_script,
-            # Empty suite is used to prevent mmdebstrap from filtering
-            # the required packages.
-            '',
+        # Pull the ostree. Ideally the eosminbase ostree would be used to
+        # minimize the buildroot, but that's not released and would fail with
+        # --use-production-ostree.
+        repo_path = config['ostree']['repodir']
+        remote = config['ostree']['remote']
+        ref = config['ostree']['ref']
+        log.info(f'Pulling OSTree ref {remote}:{ref}')
+        eib.retry(subprocess.check_call, [
+            'ostree',
+            f'--repo={repo_path}',
+            'pull',
+            remote,
+            ref,
+        ])
+
+        # Checkout the commit. The checkout is going to be mutated, so copy the
+        # objects instead of hardlinking to ensure they aren't corrupted.
+        log.info(f'Checking out OSTree ref {remote}:{ref} to {self.builddir}')
+        shutil.rmtree(self.builddir)
+        subprocess.check_call([
+            'ostree',
+            f'--repo={repo_path}',
+            'checkout',
+            '--force-copy',
```

Comment:
I checked the documentation for this argument and it says:

> Never hardlink (but may reflink if available)

Oh! for btrfs or xfs or some other COW-capable filesystem.